### ISSUE TITLE
`prefer-simple-condition-first`: `x == null` and `x != null` should be considered as "simple"

### ISSUE THREAD

**Issue by ocavue** [sindresorhus/eslint-plugin-unicorn#3558]
### Description

Currently, [prefer-simple-condition-first](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v72.0.0/docs/rules/prefer-simple-condition-first.md) classifies a "simple condition" as

> - A bare identifier (`foo`)
> - A negated simple condition (`!foo`)
> - A strict equality or inequality comparison between identifiers, `typeof identifier`, literals, signed number literals, and negative BigInt literals, with at least one identifier or `typeof identifier` (`foo === 1`, `a !== b`, `index === -1`, `typeof value === 'string'`)

It would be great if "an equality or inequality comparison between identifiers and `null`" can also be considered "simple." 
`if (x == null)` is useful for checking whether `x` is "nullish" (`null` or `undefined`).

### Examples

Given the following code

```js
let magicNumber: null | undefined | number

if (Math.random() > 0.5) {
  magicNumber = null
} else if (Math.random() > 0.5) {
  magicNumber = -1
} else if (Math.random() > 0.5) {
  magicNumber = 0
} else if (Math.random() > 0.5) { 
  magicNumber = 1 + Math.random()
}

if (magicNumber != null && magicNumber !== -1) {
  console.log(`I know ${magicNumber} must be a non-negative number`)
}
```

Current behavior: 

```js
// ❌  warning  Consider moving this simple condition first after verifying short-circuit behavior  unicorn/prefer-simple-condition-first
if (magicNumber != null && magicNumber !== -1 ) {
```

Expected behavior 

```js
// ✅
if (magicNumber != null && magicNumber !== -1 ) {
```

**Comment by ocavue** [comment]
Thanks for fixing this!
